### PR TITLE
MODINVOSTO-16  Align invoice and order sample data

### PR DESCRIPTION
**Pull request by pautau-s**
## Purpose
[MODINVOSTO-16](https://issues.folio.org/browse/MODINVOSTO-16) alignthe sample data between invoices and orders
## Approach
- added tenant API for loading sample data
- samples for invoices and invoiceLines based on orders and poLines are prepared

#### TODOS and Open Questions
- [x] Should be merged with folio-org/mod-orders-storage#122

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/main/resources/data/invoices/12345_paid.json [hunk 1]
```diff
@@ -0,0 +1,31 @@
+{
+  "id": "6b8bc989-834d-4a14-945b-4c5442ae09af",
+  "adjustments": [
+    {
+      "description": "Sales Tax",
+      "type": "Percentage",
+      "value": 5.00,
+      "prorate":"Not prorated",
```
Comment (KVupp) [inline src/main/resources/data/invoices/12345_paid.json]: Invoice level adjustments that are not prorated must have a fund distribution from which the amount will be deducted
Ref: https://github.com/folio-org/acq-models/pull/136#discussion_r287111825

File: src/main/resources/data/invoices/123invoicenumber45_approved_for_2_orders.json [hunk 2]
```diff
@@ -16,33 +16,22 @@
...
+    "313000",
+    "312325"
   ],
-  "vendorId": "168f8a63-d612-406e-813f-c7527f241ac3",
-  "manualPayment": true,
-  "metadata": {
-    "createdDate": "2018-07-19T00:00:00.000+0000",
-    "createdByUserId": "28d1057c-d137-11e8-a8d5-f2801f1b9fd1"
-  }
+  "vendorId": "c0fb5956-cdf1-11e8-a8d5-f2801f1b9fd1"
```
Comment (KVupp) [inline src/main/resources/data/invoices/123invoicenumber45_approved_for_2_orders.json]: Order 313000 has a vendor id: c0fb5956-cdf1-11e8-a8d5-f2801f1b9fd1
Order 312325 has a vendor id: 12fb63b0-cdf1-11e8-a8d5-f2801f1b9fd1

Can 2 orders with different vendor Ids roll up to 1 invoice? I don't think they can.

File: src/main/resources/data/invoice-lines/123invoicenumber45-1.json [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+      "encumbrance": "eb506834-6c70-4239-8d1a-6414a5b08007",
+      "fundId": "4428a37c-8bae-4f0d-865d-970d83d5ad55",
+      "percentage": 100.0
+    }
+  ],
+  "invoiceId": "733cafd3-895f-4e33-87b7-bf40dc3c8069",
+  "invoiceLineNumber": "123invoicenumber45-1",
+  "invoiceLineStatus": "Approved",
+  "poLineId": "90fb53f2-cdf1-11e8-a8d5-f2801f1b9fd1",
+  "productId": "0552142352",
```
Comment (KVupp) [inline src/main/resources/data/invoice-lines/123invoicenumber45-1.json]: Product Id , productIdType, subtotal doesn't match the one in PO Line

File: src/main/resources/data/invoices/123invoicenumber45_approved_for_2_orders.json (hunk 2 above)
Comment (KVupp) [reply src/main/resources/data/invoices/123invoicenumber45_approved_for_2_orders.json]: NVM. I think this is taken care of in the Orders-storage PR

File: src/main/resources/data/invoice-lines/123invoicenumber45-2_release_encumbrance.json [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+  ],
+  "invoiceId": "733cafd3-895f-4e33-87b7-bf40dc3c8069",
+  "invoiceLineNumber": "123invoicenumber45-1",
+  "invoiceLineStatus": "Approved",
+  "poLineId": "fe09e437-e487-1d59-bffb-dca466e3f2ef",
+  "productId": "3257204",
+  "productIdType": "Publisher Number",
+  "quantity": 1,
+  "releaseEncumbrance": true,
+  "subTotal": 150,
```
Comment (KVupp) [inline src/main/resources/data/invoice-lines/123invoicenumber45-2_release_encumbrance.json]: The story said the prices can be off a bit, but a question popped up here as to what happens if the discount/additional cost from PO Line is not considered. Posted it on acq-dev

File: src/main/resources/data/invoices/12345_paid.json (hunk 1 above)
Comment (pautau-s) [reply src/main/resources/data/invoices/12345_paid.json]: Done

File: src/main/resources/data/invoices/19284_paid.json [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+  "approvedBy": "b72a28f6-04fc-489f-be18-ad53a3a64f67",
+  "approvalDate": "2018-07-50T00:00:00Z",
+  "currency": "USD",
+  "folioInvoiceNo": "19284",
+  "invoiceDate": "2018-07-01T00:00:00Z",
+  "paymentMethod": "EFT",
+  "status": "Paid",
+  "source": "024b6f41-c5c6-4280-858e-33fba452a334",
+  "subTotal": 342.00,
+  "total": 352.00,
```
Comment (piotr-kalashuk) [inline src/main/resources/data/invoices/19284_paid.json]: PO with id="efff3ffe-5469-1f8f-aa3a-7afc933f9eff" this invoice is based on has 3 PO Lines with estimated total `734.37`.

File: src/main/resources/data/invoices/45723_canceled.json [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+  ],
+  "adjustmentsTotal": 0.00,
+  "currency": "USD",
+  "folioInvoiceNo": "45723",
+  "invoiceDate": "2018-07-20T00:00:00Z",
+  "paymentMethod": "EFT",
+  "status": "Cancelled",
+  "source": "4a151bd6-771e-405c-9cbe-0dcd035c58c4",
+  "subTotal": 25.00,
+  "total": 25.00,
```
Comment (piotr-kalashuk) [inline src/main/resources/data/invoices/45723_canceled.json]: PO with id="e20af8c8-b07d-47a1-9ebd-f1187e9335bd" this invoice is based on has 1 PO Line with estimated total `5960`.

File: src/main/resources/data/invoice-lines/45723-1_canceled.json [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+      "percentage": 100.0
+    }
+  ],
+  "invoiceId": "07bb89be-dd4a-42f8-bbd0-a648e14bac5d",
+  "invoiceLineNumber": "45723-1",
+  "invoiceLineStatus": "Cancelled",
+  "poLineId": "67df7bd0-ff70-2ad2-9162-54acb3e9180e",
+  "productId": "5456",
+  "productIdType": "Publisher Number",
+  "quantity": 1,
```
Comment (piotr-kalashuk) [inline src/main/resources/data/invoice-lines/45723-1_canceled.json]: PO Line has 2 electronic resources

File: src/main/resources/data/invoice-lines/82912-1.json [hunk 8]
```diff
@@ -0,0 +1,32 @@
+{
+  "id": "33dc5f4c-0387-4359-919b-9f654c757b54",
+  "adjustments": [
+    {
+      "description":"Shipping",
+      "type":"Amount",
+      "value":10.00,
```
Comment (piotr-kalashuk) [inline src/main/resources/data/invoice-lines/82912-1.json]: The PO Lin has discount 20 USD i.e. negative adjustment is expected

File: src/main/resources/data/invoices/19284_paid.json (hunk 5 above)
Comment (piotr-kalashuk) [reply src/main/resources/data/invoices/19284_paid.json]: I see, it is covering only one PO Line and 1 reseource of 2 ordered.

File: src/main/resources/data/invoice-lines/45723-1_canceled.json (hunk 7 above)
Comment (piotr-kalashuk) [reply src/main/resources/data/invoice-lines/45723-1_canceled.json]: I see, it is covering only one PO Line and 1 resource of 2 ordered.

File: src/main/resources/data/invoices/45723_canceled.json (hunk 6 above)
Comment (pautau-s) [reply src/main/resources/data/invoices/45723_canceled.json]: Updated. Now this invoice covers 1 resource of the 2 ordered .

File: src/main/resources/data/invoice-lines/82912-1.json (hunk 8 above)
Comment (pautau-s) [reply src/main/resources/data/invoice-lines/82912-1.json]: Done

File: src/main/resources/data/invoice-lines/123invoicenumber45-1.json [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+    },
+    {
+      "description": "Sales Tax",
+      "type": "Percentage",
+      "value": 5,
+      "prorate":"Not prorated",
+      "relationToTotal":"In addition to"
+    }
+  ],
+  "adjustmentsTotal": 50,
```
Comment (KVupp) [inline src/main/resources/data/invoice-lines/123invoicenumber45-1.json]: adjustment total is incorrect, must be 55

Comment (pautau-s) [reply src/main/resources/data/invoice-lines/123invoicenumber45-1.json]: Done

**Pull request merged**